src/Composer/ReleaseVersion.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace Acquia\Lightning\Composer;
+
+use Acquia\Lightning\ComponentDiscovery;
+use Composer\Script\Event;
+use Drupal\Component\Serialization\Yaml;
+
+/**
+ * A script to update the version number in Lightning's component info files.
+ */
+class ReleaseVersion {
+
+  /**
+   * Script entry point.
+   *
+   * @param \Composer\Script\Event $event
+   *   The script event.
+   */
+  public static function execute(Event $event) {
+    $arguments = $event->getArguments();
+
+    $core_package = $event->getComposer()
+      ->getRepositoryManager()
+      ->getLocalRepository()
+      ->findPackage('drupal/core', '*');
+
+    $install_path = $event->getComposer()
+      ->getInstallationManager()
+      ->getInstallPath($core_package);
+
+    $app_root = realpath($install_path . DIRECTORY_SEPARATOR . '..');
+
+    $discovery = new ComponentDiscovery($app_root);
+
+    foreach ($discovery->getAll() as $component) {
+      $path = $app_root . DIRECTORY_SEPARATOR . $component->getPathname();
+
+      $info = file_get_contents($path);
+      $info = Yaml::decode($info);
+
+      $info['version'] = reset($arguments);
```

Comment:
I tried several permutations and it does not work. The Symfony YAML component is not 100% compliant with the specification (a known issue), and it simply does not quote the version number. Trying to pass the quotes on the command line causes the quotes to either be escaped, or duplicated as the parser tries to escape them for you.

I found a workaround, though. Don't quote the version number on the command line; instead, the script will surround it with << and >> characters. This forces the YAML dumper to quote the version number; the script then strips out the << and >> stuff from the dumped YAML string. Clever me. So this is solved.